Wrapper/DBEngine.cpp

```diff
@@ -0,0 +1,273 @@
+﻿/*
+ * Copyright 2020 OmniSci, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DBEngine.h"
+#include <thrift/Thrift.h>
+#include <array>
+#include <boost/filesystem.hpp>
+#include <boost/variant.hpp>
+#include <exception>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <utility>
+#include "Catalog/Catalog.h"
+#include "Import/Importer.h"
+#include "QueryEngine/ArrowResultSet.h"
+#include "QueryEngine/CompilationOptions.h"
+#include "QueryEngine/ResultSet.h"
+#include "QueryRunner/QueryRunner.h"
+#include "Shared/Logger.h"
+#include "Shared/mapdpath.h"
+#include "Shared/sqltypes.h"
+
+namespace EmbeddedDatabase {
+
+enum class ColumnType : uint32_t { eUNK, eINT, eDBL, eFLT, eSTR, eARR };
+
+/**
+ * Cursor internal implementation
+ */
+class CursorImpl : public Cursor {
+ public:
+  CursorImpl(std::shared_ptr<ResultSet> resultSet,
+             std::shared_ptr<Data_Namespace::DataMgr> dataMgr)
+      : m_resultSet(resultSet), m_DataMgr(dataMgr) {}
+
+  size_t GetColCount() { return m_resultSet->colCount(); }
```

Comment:
Omnisci uses camelCase for method names in most places, so let's do the same.
```suggestion
  size_t getColCount() { return m_resultSet->colCount(); }
```

Reply:
ok

Reply:
fixed